src/ui/hook/useDispatchNotification.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import { shallow } from 'zustand/shallow'
+import uuid from 'short-uuid'
+import type { ActionType } from '@/ui/store'
+import { useNotificationStore } from '@/ui/store'
+import type { NotificationType } from '@/domain/notification/entity'
```

Comment:
I would have not imported this type straight from `entity` types to not leak them.

Reply:
That's a good point. But it's not an entity but a Value Object. In general, value objects in DDD don't need to have a mapping to DTOs, mainly because they are typically immutable and  don't have an identity of their own. But this is an open question.

Reply:
Yes, you are right, the `NotificationType` is not an entity itself, but it is part of the file that holds the `Notification` entity and I think it's not a good practice to directly import it if from a controller because there is no control on it, IMHO.

Reply:
Okay, that makes sense. So let's adopt a strict separation of types between layers.